l3kernel/testfiles/m3regex005.luatex.tlg

```diff
@@ -281,6 +281,11 @@ TEST 10: Caseless matching and cs
 TEST 11: Braces
 ============================================================
 |\{}|
+The token list \l_tmpa_tl contains the tokens:
+>  \{ (control sequence=\protected macro:->\ifmmode \lbrace \else \textb\ETC.)
+>  } (the letter }).
```

Comment:
caveat: comment is just based on reading the diffs not the whole code, but why is this `letter`?  if that test is a verbatim context I would have expected `the character }`  otherwise `begingroup character`

Reply:
Previously, the test did a regex replacement with replacement text `\c{{}}`.  Since `\c{...}` does not attempt to keep track of brace nesting (do we want to change that?), this produces `\{` and `}`.  With the new changes, `}` would have catcode 2 and the result would be unbalanced.  So I had to change the test somehow, and I opted for the stupid replacement text `\c{{}\cL}`, which makes the closing `}` of catcode letter.

Do we want `\c{...}` to keep track of left/right brace matching, i.e. do we want `\c{ a { } b }` to give a single control sequence?

Reply:
Don't feel competent to have a serious opinion here, my point was simply that I didn't expect to see "letter" if it has to be rendered harmless I would have expected "character", but  without considering implications I would think that it seems natural that `\c{ ... }`resulting in a single control sequence not a cs followed by some remainder. So either, yes, it should produce `\a{}b` as a single cs or it should error or the documentation for "replacement" would need ot say more than `\c{ 〈 cs name 〉 } inserts a control sequence;`.

Reply:
Guess I could test but what would `\c{ _ _ foo _ bar : } produce? A single cs?

Reply:
Yes, `\c{__foo_bar:}` gives a single cs, the only issue is when one tries to have braces inside `\c` because `{` are treated like any other characters while `}` end the csname, so `\c{ { } ... }` gives `\{` followed by some characters instead of giving a single control sequence.

Reply:
ok understood, but having braces in csnames is really weird anyway (other than in `\}` and `\{` so maybe it doesn't really matter either way as long as it is documented. Or you could have some very strange way of generating them explicitly, e.g. `\c{{}} -> \{` and `\c{}{} -> \}` .  Somewhat nicer would be `\c{{}` and `\c{}}` but that would give you unbalanced braces on the outside so not really worth it.

Reply:
Wouldn't one use `\c{\{something\}}` to match a control sequence containing braces in its name?

Reply:
@Skillmon Yes, which is why the question of what `\c{{}}` does is an unimportant implementation detail.  I'll tweak things by forbidding non-escaped `{` inside `\c{...}` in the next commit.